src/components/Verse/PlayVerseAudioButton.tsx

```diff
@@ -1,20 +1,21 @@
 import React from 'react';
 import { useDispatch } from 'react-redux';
 import { playVerse } from 'src/redux/slices/AudioPlayer/state';
+import Verse from 'types/Verse';
 
 import PlayIcon from '../../../public/icons/play-circle-outline.svg';
 import Button from '../dls/Button/Button';
 
 interface Props {
-  verseKey: string;
+  verse: Verse;
 }
 const PlayVerseAudioButton = (props: Props) => {
   const dispatch = useDispatch();
   return (
     <Button
       icon={<PlayIcon />}
       onClick={() => {
-        dispatch(playVerse(props.verseKey));
+        dispatch(playVerse(props.verse));
```

Comment:
This comment applies to the rest of the PR. I don't think that we should still be using the playVerse functionality. Probably we should deprecate it. 

What we should do instead is that we directly set the current time and simplify all the other logic.